experiments/list_unsupported.sh

```diff
@@ -34,8 +34,30 @@ done
 
 echo "$1: Unsupported features\n" > "$file_name".txt
 
+export ADA_INCLUDE_PATH="${ADA_INCLUDE_PATH:-/opt/gnat/lib/gcc/x86_64-pc-linux-gnu/4.9.4/rts-native/adainclude}"
+if [ ! -d "$ADA_INCLUDE_PATH" ]; then
+   echo >&2 "Ada include path does not exists!"
+   exit 4
+fi
+
+if ! command -v gnat > /dev/null; then
+   echo >&2 "Gnat not on PATH!"
+   exit 5
+fi
+
+export GPR_PROJECT_PATH="${GPR_PROJECT_PATH:-/opt/gnat/lib/gnat}"
+if [ ! -d "$GPR_PROJECT_PATH" ]; then
+   echo >&2 "GPR project path does not exists!"
+   exit 6
+fi
+
 for filename in $(find ${path} -name *.adb); do
    gnat2goto ${include_path} "${filename}" >>"$file_name".txt 2>&1
+   if [ "$?" -gt 0 ]; then
+      echo >&2 "Errors occurred during feature collection."
+      echo >&2 "See \"${file_name}.txt\" for details."
+      exit 7
+   fi
 done
 
 sed '/^\[/ d' < "$file_name".txt > "$file_name"_redacted.txt
```

Comment:
This is not related to this PR... But have am I right that currently this script will only report the unsupported features for the *last* file it processes, and not *all* the files it processes? In particular, we loop through all the files running gant2goto, outputting to a `${file_name}.txt` file - so one output file per input file... But we only redact and CollectUnsupported *outside* the loop....

Reply:
I should double check but I think that we append to `file_name` the result of running gnat2goto on each `filename`. Perhaps not the best choice of variable names.

Reply:
Ah, I see - that's a relief :-) I confused 'filename' and 'file_name'.